From 547b86d0a89b7a5f0483a2dff313c7127f214351 Mon Sep 17 00:00:00 2001
From: "Le.Ma" <Le.Ma@amd.com>
Date: Fri, 8 Dec 2017 10:55:19 +0800
Subject: [PATCH 4376/5855] drm/amdkcl: [4.15] Keep drm_atomic_state put/free
 on resume

This fixes patch "drm/amd/display: Do not put drm_atomic_state on resume"

Change-Id: I2f58ff653e4ee851331877cd3e40af16409fea8c
Signed-off-by: Le.Ma <Le.Ma@amd.com>
Reviewed-by: Xiaojie Yuan <Xiaojie.Yuan@amd.com>
Signed-off-by: Pavan Kumar Ramayanam <pavan.ramayanam@amd.com>
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
index 044d28c..3c4e06e 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
@@ -717,6 +717,8 @@ int amdgpu_dm_display_resume(struct amdgpu_device *adev )
 
 	ret = drm_atomic_helper_resume(ddev, adev->dm.cached_state);
 
+	drm_atomic_state_free(adev->dm.cached_state);
+
 	adev->dm.cached_state = NULL;
 
 	amdgpu_dm_irq_resume_late(adev);
-- 
2.7.4

